seedpod_ground_risk/core/plot_server.py

```diff
@@ -198,26 +198,45 @@ def compose_overlay_plot(self, x_range: Optional[Sequence[float]] = (-1.6, -1.2)
+                    for annlayer in self.annotation_layers:
+                        new_layer = FatalityRiskLayer('Fatality Risk', ac=annlayer.aircraft['name'])
+                        self.add_layer(new_layer)
+                    self.remove_duplicate_layers()
+                    self._progress_bar_callback(20)
                     self.generate_layers(bounds_poly, raster_shape)
                     self._progress_bar_callback(50)
-                    plot = Overlay([res[0] for res in self._generated_data_layers.values()])
+                    plt_lyr = list(self._generated_data_layers)[0]
+                    plot = Overlay([self._generated_data_layers[plt_lyr][0]])
                     print("Generated all layers in ", time() - t0)
                     if self.annotation_layers:
-                        plot = Overlay([res[0] for res in self._generated_data_layers.values()])
-                        raw_datas = [res[2] for res in self._generated_data_layers.values()]
-                        raster_indices = dict(Longitude=np.linspace(x_range[0], x_range[1], num=raster_shape[0]),
-                                              Latitude=np.linspace(y_range[0], y_range[1], num=raster_shape[1]))
-                        raster_grid = np.sum(
-                            [remove_raster_nans(res[1]) for res in self._generated_data_layers.values() if
-                             res[1] is not None],
-                            axis=0)
-                        raster_grid = np.flipud(raster_grid)
-                        raster_indices['Latitude'] = np.flip(raster_indices['Latitude'])
+                        plot = Overlay([self._generated_data_layers[plt_lyr][0]])
+                        res = []
+                        for dlayer in self.data_layers:
+                            raster_indices = dict(Longitude=np.linspace(x_range[0], x_range[1], num=raster_shape[0]),
+                                                  Latitude=np.linspace(y_range[0], y_range[1], num=raster_shape[1]))
+                            for keys in self._generated_data_layers:
+                                if dlayer.key == keys:
+                                    raw_data = [self._generated_data_layers[keys][2]]
+                                    raster_grid = np.sum(
+                                        [remove_raster_nans(self._generated_data_layers[keys][1])],
+                                        axis=0)
+                                    raster_grid = np.flipud(raster_grid)
+                                    raster_indices['Latitude'] = np.flip(raster_indices['Latitude'])
+
+                            for alayer in self.annotation_layers:
+                                if alayer.aircraft == dlayer.ac_dict:
+                                    res.append(alayer.annotate(raw_data, (raster_indices, raster_grid)))
 
                         self._progress_callback('Annotating Layers...')
```

Comment:
Wrong place for progress callback. Looks like the annotation is actually being done on Line 229, so it should go somewhere before this, preferably outside the loop. Alternatively, more granular user feedback could result from updating inside the loop